src/player/APVideoPlayer.js

```diff
@@ -43,8 +48,24 @@ const ReactVideoView = requireNativeComponent(
   APReactVideoView
 );
 
-const APVideoPlayer = ({ src, maxWidth, ratio, style }) => {
+const APVideoPlayer = ({
+  src: { type, object, player_configuration, startTime },
+  maxWidth,
+  ratio,
+  style
+}) => {
   const dims = getFrameDims(maxWidth, ratio);
+
+  const src = {
+    type,
+    object: Platform.select({ android: JSON.stringify(object), ios: object }),
```

Comment:
We should implement type checking before stringifying to ensure backward compatibility and make sure we don't double stringify.

Reply:
@Budaa, I have just added more specific proptypes validation to better indicate what is expected as input. I made the commit a breaking change so the version should be bumped on NPM. I don't think that supporting fuzzy proptypes makes sense, better if people remove their android work arounds when updating this module.